builder/amazon/common/step_run_spot_instance.go

```diff
@@ -291,7 +290,7 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
 				// we can wait on those operations, this can be removed.
 				return true
 			}
-			return request.IsErrorRetryable(err)
+			return false
```

Comment:
Good! I'll change that. I had a hard time testing this, thanks for running the tests!

Reply:
Yes good catch Wilken, I didn't read the `IsErrorRetryable` code when doing this; I think it would be a good idea to have errors be not retryable by default.

Reply:
`ec2conn.CreateFleet` returns an array of error and we flat it into a single error to use it in the retry logic. Because of that, is not possible to use `IsAWSErr` because the error won't be identified as a `awserr.Error`.  So checking if the string contains the iamInstanceProfile error was the right way for the previous code